backend/capellacollab/core/exceptions.py

```diff
@@ -3,59 +3,39 @@
 
 import dataclasses
 
-import fastapi
-from fastapi import exception_handlers, status
+from fastapi import status
+
+
+@dataclasses.dataclass(kw_only=True)
+class BaseError(Exception):
+    status_code: int
+    title: str
+    reason: str
+    err_code: str | None = None
 
 
 @dataclasses.dataclass
-class ExistingDependenciesError(Exception):
+class ExistingDependenciesError(BaseError):
     entity_name: str
     entity_type: str
     dependencies: list[str]
 
-
-async def existing_dependencies_exception_handler(
-    request: fastapi.Request, exc: ExistingDependenciesError
-) -> fastapi.Response:
-    dependencies_str = ", ".join(exc.dependencies)
-    return await exception_handlers.http_exception_handler(
-        request,
-        fastapi.HTTPException(
-            status_code=status.HTTP_409_CONFLICT,
-            detail={
-                "reason": [
-                    f"The {exc.entity_type} '{exc.entity_name}' can not be deleted. Please remove the following dependencies first: {dependencies_str}"
-                ]
-            },
-        ),
-    )
+    def __post_init__(self):
+        self.status_code = status.HTTP_409_CONFLICT
+        self.title = f"{self.entity_name} cannot be deleted"
+        self.reason = f"""The {self.entity_type} '{self.entity_name}' can not be deleted.
+        Please remove the following dependencies first: {", ".join(self.dependencies)}
+        """
```

Comment:
```suggestion
        self.reason = (
          f"The {self.entity_type} '{self.entity_name}' can not be deleted."
          f"Please remove the following dependencies first: {", ".join(self.dependencies)}"
        )
```

Avoid multi-line strings. They have leading whitespaces. Instead use the proposed format.